include/xgboost/data.h

```diff
@@ -410,7 +426,7 @@ class DMatrix {
    *     This can be nullptr for common cases, and in-memory mode will be used.
    * \return a Created DMatrix.
    */
-  static DMatrix* Create(std::unique_ptr<DataSource>&& source,
+  static DMatrix* Create(std::unique_ptr<DataSource<SparsePage>>&& source,
```

Comment:
Do we need to add a different `Create` method for other type of pages?

Reply:
I guess that depends on where we want to take this. For now I'm still assuming CSR is the default format, and other formats are lazily generated on the fly. In the future we could move towards parsing input a desired format directly.